CRM/Contribute/Form/Contribution.php

```diff
@@ -1338,7 +1338,7 @@ protected function submit($submittedValues, $action, $pledgePaymentID) {
     // reassign submitted form values if the any information is formatted via beginPostProcess
     $submittedValues = $this->_params;
 
-    if (!empty($submittedValues['price_set_id']) && $action & CRM_Core_Action::UPDATE) {
+    if ($this->getPriceSetID() && $action & CRM_Core_Action::UPDATE) {
```

Comment:
OK, just verbalizing. The ordinary caller is

```php
    $submittedValues = $this->controller->exportValues($this->_name);

    try {
      $contribution = $this->submit($submittedValues, $this->_action, $this->_ppID);
```

 The idea is that either way (before/after patch) the data flows from `$this-controller->exportValues($this->_name`) -- just a different path. Seems reasonable.

There is another flow going through `testSubmit()`:

```php
  public function testSubmit($params, $action, $creditCardMode = NULL) {
    ...
    return $this->submit(array_merge($defaults, $params), ...);
  }
```

That flow might make a difference for `testSubmit(...)`-based flows. But `testSubmit()` is only used by unit-tests. I think we're OK here.

Reply:
@totten yeah so the history of `testSubmit` is we didn't originally have a good way to imitate the way the form really worked so we added hacky functions to do that - however, we have usable methods now.

Reply:
@eileenmcnaughton Makes sense.

There's a merge-conflict now. But otherwise MOP.